### PR TITLE
Erase Todo functionality, create all BE functionality in backlog Membuat Akun in Epic01, create model AppUser and change the migrations

### PR DESCRIPTION
**Pull request by JohannesSetiawan**
Things that changes:
1. All Todo functionality were removed (Todo models, the views, and the urls)
2. Add django_rest_framework and simplejwt to requirements.txt. Also change both settings.py to accomodate rest_framework and simplejwt. Also add both of it to installed apps.
3. Create model AppUser and set that model to be the AUTH_USER_MODEL in both settings.py. Also change the migrations due to conflict with the previous migrations.
4. Create functionality for users register and log in using JWT tokens and rest-framework. Also test both of the functionality with happy and unhappy test case. Tested both APIs on Postman.
5. Create new workflows that runs Build and Test for all branches
6. Add environtment variables and variables in settings.py and all workflows for SendVerificationEmail and SendRecoverPasswordEmail feature. Also add new secrets on GitHub.
7. Implements SendVerificationEmail and SendRecoverPasswordEmail feature. and tests. Test both feature on Postman.

### CODE REVIEW

File: authentication/urls.py [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+from rest_framework_simplejwt.views import TokenObtainPairView
+from rest_framework_simplejwt.views import TokenRefreshView
 
 app_name = 'authentication'
 
 urlpatterns = [
-    path('try', returnTodoJson, name='returnTodoJson'),
+    path('register/', RegisterView.as_view(), name='register'),
+    path('login/', LoginView.as_view(), name='login'),
+    path('token/', TokenObtainPairView.as_view(), name='token_obtain_pair'),
```
Comment (emiriko) [inline authentication/urls.py:11]: Personally, I don't think this is necessary. I think the best apporach is to send token after POST request on login endpoint (which you already did), so probably just remove this one.

We will just need to use refresh token endpoint to refresh our access token (which I expect refresh token endpoint will send the access token back to us).

File: authentication/tests.py [hunk 2]
```diff
@@ -1,10 +1,132 @@
...
+        data = {
+                "username":"user1",
+                "password":"pass1",
+                "email":"email1@email.com"
+        }
+        response = self.client.post(REGISTER_LINK, json.dumps(data), content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        response = self.client.post(REGISTER_LINK, json.dumps(data), content_type='application/json')
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json()['msg'],"Username and/or email already taken!")
```
Comment (emiriko) [inline authentication/tests.py:60]: Perhaps you could add if the some request body are missing, such as username, etc.

**Review by emiriko — CHANGES_REQUESTED** [review]

File: authentication/tests.py (hunk 2 above)
Comment (JohannesSetiawan) [reply authentication/tests.py:60]: The test_empty_input test that though. I also think that at first but it feels like duplicating the test. So i guess one test is enough

File: authentication/urls.py (hunk 1 above)
Comment (JohannesSetiawan) [reply authentication/urls.py:11]: Ok, fix it real soon

File: authentication/tests.py (hunk 2 above)
Comment (emiriko) [reply authentication/tests.py:60]: Oh, I meant what if the request body is like this:

{
 "username":"user1",
 "password":"pass1",
}

Note that there is no email attribute there.

**Review by emiriko — APPROVED** [review]

**Pull request merged**